GeoLib/Polyline.h

```diff
@@ -50,6 +51,109 @@ enum class Location
 class Polyline : public GeoObject
 {
 public:
+	class SegmentIterator
+	    : public std::iterator<std::forward_iterator_tag, LineSegment>
```

Comment:
The iterator class is part of the interface of the polyline and has to declared in the header file. I moved the implementation to the cpp file. Now the number of lines for the class declaration is only 35 lines compared to 100 lines before (declaration + implementation).